sdk/eventhub/eventhubs-checkpointstore-table/src/tableCheckpointStore.ts

```diff
@@ -3,11 +3,29 @@
 
 import { CheckpointStore, PartitionOwnership, Checkpoint } from "@azure/event-hubs";
 
+import { odata, TableClient } from "@azure/data-tables";
+
+export interface customCheckpoint extends Checkpoint {
```

Comment:
You want me to rename CustomCheckpoint to CheckpointEntity? or you want me to rename the variable itself to CheckpointEntity.

Also if I dont extend the Checkpoint there I wont be able to store Checkpoints in the table which needs partitionKey and rowKey. I will add the @internal doc tag too.

Reply:
Let me take a step back. What fields do you actually need to store for a CheckpointEntity?

You have the `partitionKey` and the `rowKey`. `rowKey` is always `partitionId`, so you don't need to separately store the partitionId. Checkpoint defines `sequenceNumber`, but in Tables we should store this as `sequencenumber` so that all languages can interop together.

You also don't need to store the namespace, event hub name, or consumer group explicitly because they user passes those in to every public method. They have to pass those parameters to the functions so that we can construct the partition key.

Remember that the entity you're saving to Tables doesn't have to match exactly with the Checkpoint/Ownership the user passes to you.